### PR TITLE
Add missing documentation for JVM metrics in Pulsar broker

### PR DESCRIPTION
**Pull request by asafm**
### Documentation

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@mattisonchao Could you please review this PR from a technical perspective? Then writers will review it from a technical writing side. Thank you!

**Review by mattisonchao — APPROVED** [review]

File: docs/reference-metrics.md [hunk 1]
```diff
@@ -600,6 +600,67 @@ All the metadata store metrics are labeled with the following labels:
...
+| jvm_memory_objects_pending_finalization | Gauge | The number of objects waiting in the finalizer queue. |
+| jvm_memory_bytes_used | Gauge | Used bytes of a given JVM memory area. |
+| jvm_memory_bytes_committed | Gauge | Committed (bytes) of a given JVM memory area. |
+| jvm_memory_bytes_max | Gauge | Max (bytes) of a given JVM memory area. |
+| jvm_memory_bytes_init | Gauge | Initial bytes of a given JVM memory area. |
+| jvm_memory_pool_bytes_used | Used bytes of a given JVM memory pool. |
+| jvm_memory_pool_bytes_committed | Gauge | Committed bytes of a given JVM memory pool. |
+| jvm_memory_pool_bytes_max | Gauge | Max bytes of a given JVM memory pool. |
+| jvm_memory_pool_bytes_init | Gauge | Initial bytes of a given JVM memory pool. |
+| jvm_memory_pool_collection_used_bytes | Gauge | Used bytes after last collection of a given JVM memory pool. |
```
Comment (Anonymitaet) [inline docs/reference-metrics.md:627]:
```suggestion
| jvm_memory_pool_collection_used_bytes | Gauge | Used bytes after the last collection of a given JVM memory pool. |
```

File: docs/reference-metrics.md [hunk 2]
```diff
@@ -600,6 +600,67 @@ All the metadata store metrics are labeled with the following labels:
...
+| jvm_memory_pool_collection_committed_bytes | Gauge | Committed after last collection bytes of a given JVM memory pool. |
+| jvm_memory_pool_collection_max_bytes | Gauge | Max bytes after last collection of a given JVM memory pool. |
```
Comment (Anonymitaet) [inline docs/reference-metrics.md:629]:
```suggestion
| jvm_memory_pool_collection_max_bytes | Gauge | Max bytes after the last collection of a given JVM memory pool. |
```

Comment (Anonymitaet) [reply docs/reference-metrics.md:629]: Same comments for versioned docs.

**Review by Anonymitaet — COMMENTED** [review]
Thanks for your contribution! Have you previewed your changes and ensured everything goes as expected? 

If not, please [preview your changes locally](https://pulsar.apache.org/contribute/document-preview/#preview-changes) and attach the screenshots in this PR. In this way, you can get your PR merged more quickly. Thanks!

**Comment by asafm** [comment]
image
image
image
image
image
image

**Comment by asafm** [comment]
@Anonymitaet I don't know how to make the name column bigger

**Review by Anonymitaet — APPROVED** [review]

**Comment by Anonymitaet** [comment]
> @Anonymitaet I don't know how to make the name column bigger

No worries, that's fine.

Feel free to ping me to merge if all tests pass.

**Comment by asafm** [comment]
All tests pass @Anonymitaet

**Comment by tisonkun** [comment]
@asafm Thank you!

**Pull request merged**